CoreWebServer/src/au/gov/asd/tac/constellation/webserver/resources/constellation_client.py

```diff
@@ -564,23 +564,45 @@ def open_graph(self, filename):
 
         return graph['id']
 
-    def new_graph(self, schema_name=None):
-        """Open a new graph using the given schema.
+    def new_graph(self, schema_name=None, graph_name=None):
+        """Open a new graph using the given schema and optional graph name.
 
         The default schema is whatever CONSTELLATION's default schema is.
+        If the graph name is left blank, the default name `analytic graph1` etc. will be used.
 
         :param schema_name: The optional schema of the new graph.
+        :param graph_name: The optional graph name for the new graph.
 
         :returns: The id of the new graph.
         """
 
         params = {}
         if schema_name:
             params['schema_name'] = schema_name
+        if graph_name:
+            params['graph_name'] = graph_name
         graph = self.call_service('new_graph', verb='post', args=params).json()
 
         return graph['id']
 
+    def rename_graph(self, graph_id=None, new_graph_name):
+        """Rename a graph. If a graph is not specified, the active graph is renamed.
+
+        :param graph_id: The optional id of the graph to rename.
+        :param new_graph_name: The new name for the graph.
+
+        :returns: The id, previous name and the new name of the graph.
+        """
+
+        params = {}
+        if graph_id:
+            params['graph_id'] = graph_id
+        
+        params['new_graph_name'] = new_graph_name
+        graph = self.call_service('new_graph', verb='post', args=params).json()
+
+        return graph['id', 'previous_name', 'new_name']
```

Comment:
This line causes an error during testing ...
It doesn't like having multiple field names listed.
We can simply replace the line with:
`return graph`
This will return all the fields in the graph object in json format.
(which happens to be only those 3 fields that were listed)